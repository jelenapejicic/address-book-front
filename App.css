.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  margin: auto;
  
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

h1 {
  font-family: 'Open Sans Condensed', sans-serif;
  font-size: 40px;
  text-align: center;

}

.add-button {
  min-width: 50px;
  width: auto;
  height: 50px;
  letter-spacing: 0.5px;
  line-height: 50px;
  padding: 0 35px 0 35px;
  font-size: 15px;
  background-color: black;
  color: white;
  text-transform: uppercase;
  font-family: 'Open Sans Condensed';
  font-weight: bolder;
  border: none;
  cursor: pointer;
  border-radius: 5px;
  margin-left: 30px;

}

.add-button:hover {
  background-color: rgb(206, 211, 209);
  color: black;
  border: 1px solid gray;
}
.contact-box{
 width: 85%;
}
.edit-button{
  border: 0px;
  text-align: right;
  padding: 0px;
  background-color: rgb(206, 211, 209);
  font-weight: bold;
}