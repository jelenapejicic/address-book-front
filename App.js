import React, { useState, useEffect } from 'react';
import './App.css';
import { ContactList } from './components/contast-list/contact-list.component';
import { SearchBox } from './components/search-box/search-box.component'
import {CreateContact} from './components/create-contact/create-contact.component';
import Popup from "reactjs-popup";


const App = () => {

  const [contacts, setContacts] = useState([]);
  const [searchTerm, setSearchTerm] = useState('');

  const Modal = (props) => (
    <Popup className="modalll"
      trigger={<button className={props.className}> <span className={props.iconClass}></span> </button>}
      modal
      closeOnDocumentClick
      >
      <CreateContact handleState={handleState} contactDetails={props.contactDetails}></CreateContact>
    </Popup>
  );
  
  
  useEffect(() => {

  const fetchFunc = async () => {
      const response = await fetch(
        'http://localhost:8080/imenik/api/contact/read.php'
      );
      const resJson = await response.json();
      console.log(resJson.records);

      if(resJson.records !== undefined){
        const filtered = resJson.records.filter(contact =>
          contact.name.toLowerCase().includes(searchTerm.toLowerCase())
        );
        setContacts(filtered);
      } 
      
    };
    fetchFunc();
  }, [searchTerm]);

  
  const handleState = () => {
   
    const fetchFunc = async () => {
      const response = await fetch(
        'http://localhost:8080/imenik/api/contact/read.php'
      );
      const resJson = await response.json();
      console.log(resJson.records);
      setContacts(resJson.records);
    };
    fetchFunc();
  }


  return (
    <div >
      <h1>Address Book</h1>
      <div className="contact-box">
      <SearchBox
        placeholder="Search names..."
        handleChange={e => setSearchTerm(e.target.value)}
      />
      <Modal className="add-button" iconClass="fa fa-plus"/>
      </div>
      <ContactList contacts={contacts} handleState={handleState} modal={Modal}></ContactList>
    
    </div>
  )
}

export default App;

