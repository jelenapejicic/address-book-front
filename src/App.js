import React, { useState, useEffect } from 'react';
import './App.css';
import { ContactList } from './components/contast-list/contact-list.component';
import { SearchBox } from './components/search-box/search-box.component'
import { CreateContact } from './components/create-contact/create-contact.component';
import Popup from "reactjs-popup";
import { fetchFunc } from "./Api";

const App = () => {

  const [contacts, setContacts] = useState([]);
  const [searchTerm, setSearchTerm] = useState('');

  const Modal = (props) => (
    <Popup className="modalll"
      trigger={<button className={props.className}> <span className={props.iconClass}></span> </button>}
      modal
      closeOnDocumentClick
    >
      <CreateContact handleState={handleState} contactDetails={props.contactDetails}></CreateContact>
    </Popup>
  );


  useEffect(() => {

    fetchFunc('read.php', null).then(data => {

      if (data !== undefined) {
        const filtered = data.filter(contact =>
          contact.name.toLowerCase().includes(searchTerm.toLowerCase())
        );
        setContacts(filtered);
      }
    });

  }, [searchTerm]);


  const handleState = () => {

    fetchFunc('read.php', null).then(data => setContacts(data));

  }
  
  return (
    <div >
      <h1>Address Book</h1>
      <div className="contact-box">
        <SearchBox
          placeholder="Search names..."
          handleChange={e => setSearchTerm(e.target.value)}
        />
        <Modal className="add-button" iconClass="fa fa-plus" />
      </div>
      <ContactList contacts={contacts} handleState={handleState} modal={Modal}></ContactList>

    </div>
  )
}

export default App;

