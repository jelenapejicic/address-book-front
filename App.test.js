import React from 'react';
import * as enzyme from 'enzyme';
import { fetchFunc } from './App';
import App from './App';
//import "babel-polyfill";
// import { fetchData } from './api';
// import { act } from 'react-dom/test-utils';


it('it should render App component', () => {

    const wrapper = enzyme.shallow(<App/>);
    expect(wrapper).toMatchSnapshot();

});

it('it calls api to get people' , async () => {

    // let data = await fetchFunc();
    // expect(data).toBe('peanut butter');

    expect.assertions(1);
    return expect(fetchFunc()).rejects.toEqual({
    error: 'User with 3 not found.',
  });

} );

