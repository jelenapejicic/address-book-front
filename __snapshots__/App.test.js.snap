// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render App component 1`] = `ShallowWrapper {}`;
